src/main/java/org/intellij/sdk/codesync/Utils.java

```diff
@@ -395,9 +400,101 @@ public static void ChangesHandler(DocumentEvent event, Project project) {
+        SimpleDateFormat pattern = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");
+        pattern.setTimeZone(TimeZone.getTimeZone("UTC"));
+        return pattern.format(date.toMillis());
+    }
+
+    public static Boolean isDirectoryDelete(String repoPath, String branch, String relativeFilePath) {
+        String shadowPath = String.format("%s/%s/%s/%s", SHADOW_REPO, repoPath, branch, relativeFilePath);
+        File shadowPathFile = new File(shadowPath);
+
+        return shadowPathFile.exists() && shadowPathFile.isDirectory();
+    }
+
+    public static Map<String, Object> getFileInfo(String filePath) throws FileInfoError {
+        Map<String, Object> fileInfo = new HashMap<>();
+        File file = new File(filePath);
+        Path path = file.toPath();
+        try {
+            BasicFileAttributes fileAttributes = Files.readAttributes(path, BasicFileAttributes.class);
+            fileInfo.put("size", fileAttributes.size());
+            fileInfo.put("creationTime", formatDate(fileAttributes.creationTime()));
+            fileInfo.put("modifiedTime", formatDate(fileAttributes.lastModifiedTime()));
+            fileInfo.put("isBinary", isBinaryFile(file));
+        } catch (IOException error) {
+            throw new FileInfoError(error.getMessage());
+        }
+
+        return fileInfo;
+    }
+
+    public static String computeDiff(String initialVersion, String latterVersion) {
+        diff_match_patch dmp = new diff_match_patch();
+        LinkedList<diff_match_patch.Patch> patches = dmp.patch_make(initialVersion, latterVersion);
+
+        // return text representation of patches objects
+        return dmp.patch_toText(patches);
+    }
+
+    public static boolean isBinaryFile(File f) throws IOException {
+        try {
+            return !isTextFile(f);
+        } catch (Exception e){
+            return false;
+        }
+
+    }
+
+    public static boolean getBoolValue(Map<String, Object> map, String key, boolean defaultValue) {
+        Boolean binaryValue = (Boolean) map.getOrDefault(key, defaultValue);
+        return (binaryValue != null ? binaryValue: false);
+    }
+
+    private static boolean isTextFile(File f) throws Exception {
+        if(!f.exists())
+            return false;
+        FileInputStream in = new FileInputStream(f);
+        int size = in.available();
+        if(size > 1000)
+            size = 1000;
+        byte[] data = new byte[size];
+        in.read(data);
+        in.close();
+        String s = new String(data, "ISO-8859-1");
+        String s2 = s.replaceAll(
+                "[a-zA-Z0-9ßöäü\\.\\*!\"§\\$\\%&/()=\\?@~'#:,;\\"+
+                        "+><\\|\\[\\]\\{\\}\\^°²³\\\\ \\n\\r\\t_\\-`´âêîô"+
+                        "ÂÊÔÎáéíóàèìòÁÉÍÓÀÈÌÒ©‰¢£¥€±¿»«¼½¾™ª]", "");
+        // will delete all text signs
+
+        double d = (double)(s.length() - s2.length()) / (double)(s.length());
+        // percentage of text signs in the text
+        return d > 0.95;
```

Comment:
Maybe a constant for threshold value

Reply:
I think this way code is more readable. We will not be using it anywhere else either so hardcoded value makes more sense here.

Reply:
sure, your call, was just a minor observation